segmentstore/server/src/test/java/io/pravega/segmentstore/server/containers/DebugStreamSegmentContainerTests.java

```diff
@@ -0,0 +1,304 @@
+import io.pravega.segmentstore.server.OperationLogFactory;
+import io.pravega.segmentstore.server.ReadIndexFactory;
+import io.pravega.segmentstore.server.SegmentContainer;
+import io.pravega.segmentstore.server.SegmentContainerFactory;
+import io.pravega.segmentstore.server.WriterFactory;
+import io.pravega.segmentstore.server.attributes.AttributeIndexConfig;
+import io.pravega.segmentstore.server.attributes.AttributeIndexFactory;
+import io.pravega.segmentstore.server.attributes.ContainerAttributeIndexFactoryImpl;
+import io.pravega.segmentstore.server.logs.DurableLogConfig;
+import io.pravega.segmentstore.server.logs.DurableLogFactory;
+import io.pravega.segmentstore.server.reading.ContainerReadIndexFactory;
+import io.pravega.segmentstore.server.reading.ReadIndexConfig;
+import io.pravega.segmentstore.server.tables.ContainerTableExtension;
+import io.pravega.segmentstore.server.tables.ContainerTableExtensionImpl;
+import io.pravega.segmentstore.server.writer.StorageWriterFactory;
+import io.pravega.segmentstore.server.writer.WriterConfig;
+import io.pravega.segmentstore.storage.AsyncStorageWrapper;
+import io.pravega.segmentstore.storage.DurableDataLogFactory;
+import io.pravega.segmentstore.storage.SegmentRollingPolicy;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.cache.CacheStorage;
+import io.pravega.segmentstore.storage.cache.DirectMemoryCache;
+import io.pravega.segmentstore.storage.mocks.InMemoryDurableDataLogFactory;
+import io.pravega.segmentstore.storage.mocks.InMemoryStorage;
+import io.pravega.segmentstore.storage.mocks.InMemoryStorageFactory;
+import io.pravega.segmentstore.storage.rolling.RollingStorage;
+import io.pravega.shared.segment.SegmentToContainerMapper;
+import io.pravega.test.common.ThreadPooledTestSuite;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+import java.io.ByteArrayInputStream;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static io.pravega.segmentstore.server.containers.ContainerRecoveryUtils.recoverAllSegments;
+
+/**
+ * Tests for DebugStreamSegmentContainer class.
+ */
+@Slf4j
+public class DebugStreamSegmentContainerTests extends ThreadPooledTestSuite {
+    private static final int MIN_SEGMENT_LENGTH = 0; // Used in randomly generating the length for a segment
+    private static final int MAX_SEGMENT_LENGTH = 10100; // Used in randomly generating the length for a segment
+    private static final int CONTAINER_ID = 1234567;
+    private static final int EXPECTED_PINNED_SEGMENT_COUNT = 1;
```

Comment:
Is this needed? 
Delete constants not used.

Reply:
Deleted.